### PR TITLE
Add support for MiTAC MiTAC-LY1200-B32H0-C3 platform

### PR DESCRIPTION
**Pull request by staneshieh**
**- What I did**
Add MiTAC-LY1200-B32H0-C3.
**- How I did it**
Add drv code for sfp, eeprom, psu utility tool
**- How to verify it**
Test cmd. cat/echo sysfs, spfutil, psu,  eeprom, sensors to test.
**- Description for the changelog**
Add MiTAC-LY1200-B32H0-C3 drv code and utility

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	url = https://github.com/Azure/sonic-swss-common
 [submodule "sonic-linux-kernel"]
 	path = src/sonic-linux-kernel
-	url = https://github.com/Azure/sonic-linux-kernel
+	url = https://github.com/MiTAC-EBU/sonic-linux-kernel.git
```
Comment (lguohan) [inline .gitmodules]: you probably do not want to change this.

File: device/mitac/x86_64-mitac_ly1200_b32h0_c3-r0/update_data_ram [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		BCM_IF_ERROR_RETURN(bcm_linkscan_register(unit, link_callback));
+		BCM_IF_ERROR_RETURN(bcm_port_config_get(unit, &config));
+		BCM_IF_ERROR_RETURN(bcm_link_change(unit, config.port));
+	} else {
+		BCM_IF_ERROR_RETURN(bcm_linkscan_unregister(unit, link_callback));
+	}
+	return BCM_E_NONE;
+}
+
+link_monitor(0,1);
```
Comment (lguohan) [inline device/mitac/x86_64-mitac_ly1200_b32h0_c3-r0/update_data_ram]: why do you need such bcm code? what is the purpose?

Comment (staneshieh) [reply device/mitac/x86_64-mitac_ly1200_b32h0_c3-r0/update_data_ram]: We want to use one led_proc_init.soc to support different speeds for front panel port physical connections.

The advantage is that as long as SONiC supports port speed changes on Broadcom platform, we don't need to update led_proc_init.soc.

Comment (lguohan) [reply device/mitac/x86_64-mitac_ly1200_b32h0_c3-r0/update_data_ram]: understand the requirement, however, we cannot have bcm api call in our repo. for now, we have to remove them.

File: .gitmodules (hunk 1 above)
Comment (staneshieh) [reply .gitmodules]: I have restored the contents of .gitmodules to keep Azure sonic linux kernel and opened a new pull request(https://github.com/Azure/sonic-linux-kernel/pull/39) for my patches.

File: device/mitac/x86_64-mitac_ly1200_b32h0_c3-r0/update_data_ram (hunk 2 above)
Comment (staneshieh) [reply device/mitac/x86_64-mitac_ly1200_b32h0_c3-r0/update_data_ram]: Would you accept if we moved it to our repo (${sonic}/platform/broadcom/sonic-platform-modules-mitac/)?

Comment (lguohan) [reply device/mitac/x86_64-mitac_ly1200_b32h0_c3-r0/update_data_ram]: To do this in the right way, if you can propose right SAI API to control the port LED in the SAI community, we can ask the asic vendor to implement it.

Comment (staneshieh) [reply device/mitac/x86_64-mitac_ly1200_b32h0_c3-r0/update_data_ram]: understand the explanation, we will remove it. I will close this PR and open a new PR when we finish modifications. Thanks a lot.

**Pull request closed without merging**